flows/parent_flow.py

```diff
@@ -1,47 +1,367 @@
+                    name="Docker flow/launch_docker",
+                    parameters=deployment_data
+                )
+                
+                # Check the status of the flow run
+                flow_run = await client.read_flow_run(flow_run.id)
+                
+                if flow_run.state.is_failed():
+                    prefect_logger.error(f"Step {i+1} failed with state: {flow_run.state.type}")
+                    return Failed(message=f"Child flow failed with state: {flow_run.state.type}")
+                    
+                flow_run_id = str(flow_run.id)
+                
+            elif target_env == FlowType.podman:
+                # Prepare podman parameters
+                podman_relevant_params = {
+                    "image_name": algorithm_details["image_name"],
+                    "image_tag": algorithm_details["image_tag"],
+                    "command": f"python {python_file}",
+                    "volumes": json.loads(algorithm_details["volumes"]),
+                    "network": algorithm_details["network"],
+                    "env_vars": {},
+                    "params": params
+                }
+                # Validate parameters with the schema
+                podman_params = PodmanParams(**podman_relevant_params)
+                # If there's a previous flow run ID, set it in the parameters
+                if flow_run_id:
+                    if "io_parameters" not in podman_params.params:
+                        podman_params.params["io_parameters"] = {}
+                    podman_params.params["io_parameters"]["uid_retrieve"] = flow_run_id
+                
+                # Run the podman deployment with parameters
+                deployment_data = {
+                    "podman_params": podman_params.dict(),
+                    "prev_flow_run_id": flow_run_id
+                }
+                flow_run = await run_deployment(
+                    name="Podman flow/launch_podman", 
+                    parameters=deployment_data
+                )
+                
+                # Check the status of the flow run
+                flow_run = await client.read_flow_run(flow_run.id)
+                
+                if flow_run.state.is_failed():
+                    prefect_logger.error(f"Step {i+1} failed with state: {flow_run.state.type}")
+                    return Failed(message=f"Child flow failed with state: {flow_run.state.type}")
+                    
+                flow_run_id = str(flow_run.id)
+                
+            elif target_env == FlowType.slurm:
+                # Prepare slurm parameters
+                slurm_relevant_params = {
+                    "job_name": f"{model_name}_{task_name}",
+                    "num_nodes": algorithm_details["num_nodes"],
+                    "partitions": json.loads(algorithm_details["partitions"]),
+                    "reservations": json.loads(algorithm_details["reservations"]),
```

Comment:
This could also be part of the config file depending on the SLURM resources that are available